x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/List/index.tsx

```diff
@@ -135,98 +174,48 @@ export function AgentConfigurationList({
-          ),
-          icon: 'pencil',
-          color: 'primary',
-          type: 'icon',
-          onClick: (config: Config) => {
-            setSelectedConfig(config);
-            setIsFlyoutOpen(true);
-          }
-        }
-      ]
+      render: (config: Config) => (
+        <EuiButtonIcon
+          aria-label="Edit"
+          iconType="pencil"
+          href={editAgentConfigurationHref(config.service)}
+        />
+      )
+    },
+    {
+      width: px(units.double),
+      name: '',
+      render: (config: Config) => (
+        <EuiButtonIcon
+          aria-label="Delete"
+          iconType="trash"
+          onClick={() => setConfigToBeDeleted(config)}
+        />
+      )
     }
   ];
 
-  const emptyStatePrompt = (
-    <EuiEmptyPrompt
-      iconType="controlsHorizontal"
-      title={
-        <h2>
-          {i18n.translate(
-            'xpack.apm.settings.agentConf.configTable.emptyPromptTitle',
-            { defaultMessage: 'No configurations found.' }
-          )}
-        </h2>
-      }
-      body={
-        <>
-          <p>
-            {i18n.translate(
-              'xpack.apm.settings.agentConf.configTable.emptyPromptText',
-              {
-                defaultMessage:
-                  "Let's change that! You can fine-tune agent configuration directly from Kibana without having to redeploy. Get started by creating your first configuration."
-              }
-            )}
-          </p>
-        </>
-      }
-      actions={
-        <EuiButton color="primary" fill onClick={() => setIsFlyoutOpen(true)}>
-          {i18n.translate(
-            'xpack.apm.settings.agentConf.configTable.createConfigButtonLabel',
-            { defaultMessage: 'Create configuration' }
-          )}
-        </EuiButton>
-      }
-    />
-  );
-
-  const failurePrompt = (
-    <EuiEmptyPrompt
-      iconType="alert"
-      body={
-        <>
-          <p>
-            {i18n.translate(
-              'xpack.apm.settings.agentConf.configTable.configTable.failurePromptText',
-              {
-                defaultMessage:
-                  'The list of agent configurations could not be fetched. Your user may not have the sufficient permissions.'
-              }
-            )}
-          </p>
-        </>
-      }
-    />
-  );
-
-  if (status === 'failure') {
-    return failurePrompt;
-  }
-
-  if (status === 'success' && isEmpty(data)) {
-    return emptyStatePrompt;
-  }
-
   return (
-    <ManagedTable
-      noItemsMessage={<LoadingStatePrompt />}
-      columns={columns}
-      items={data}
-      initialSortField="service.name"
-      initialSortDirection="asc"
-      initialPageSize={50}
-    />
+    <>
+      {configToBeDeleted ? (
+        <ConfirmDeleteModal
+          config={configToBeDeleted}
+          onCancel={() => setConfigToBeDeleted(null)}
+          onConfirm={() => {
+            setConfigToBeDeleted(null);
+            refetch();
+          }}
+        />
+      ) : null}
```

Comment:
I believe you don't need to return null.
 ```suggestion
      {configToBeDeleted && (
        <ConfirmDeleteModal
          config={configToBeDeleted}
          onCancel={() => setConfigToBeDeleted(null)}
          onConfirm={() => {
            setConfigToBeDeleted(null);
            refetch();
          }}
        />
      )}
```